### PR TITLE
Added delete method to file adapters (closes #1948)

### LINKED ISSUE #1948: Provide a `delete` method on file adapters
Both the local file adapter and the cloudinary file adapter currently persist files forever (there is no automatic or manual deletion phase).

Adding an automatic deletion phase is tricky and a problem we haven't thought through yet.

However, adding a manual deletion phase which a Keystone App developer opts-in to could be a great half-way point.

This idea originated [from the Keystone Slack](https://keystonejs.slack.com/archives/C02FLV1JG/p1573246847221200) by @Vultraz:

> should [deleting files] perhaps be something handled by the adapters, or is it better to leave it to people to use hooks?)
>
> for the local file case I've found the hooks are pretty easy to set up. Just add a `afterDelete` list hook and `beforeChange` field hook and it works like a breeze
>
> here's the full function I use
> ```
> const deleteFileOnDisk = async ({ existingItem: { file } = {} }) => {
>   if (file) {
>     fs.unlink(path.join(fileAdapter.src, file.filename), (err) => {
 >      if (err) {
 >        console.log(err);
>       }
>     });
>   }
> }
> ```
> 
> I know KS4 had a `removeFile` method on its local file storage adapter
>
> I was thinking that might be something to add back? The local file case is quite simple, but I imagine it might be harder to roll a custom delete impl with the CloudinaryFileAdapter since you'd need to know their API 
> 
> So, I was thinking something along the lines of..
> 
> ```
>   hooks: {
>     afterDelete: ({ existingItem }) => adapter.deleteFile(existingItem)
>   }
> ```
> 
> that way, the deletion implementation gets abstracted away but the end user can still choose if, where, and when to deploy the deletion

### PR DESCRIPTION
**Pull request by Vultraz**
Usage example (included in README):
```
keystone.createList('UploadTest', {
  fields: {
    file: {
      type: File,
      adapter: fileAdapter,
      hooks: {
        beforeChange: ({ existingItem = {} }) => fileAdapter.delete(existingItem),
      },
    },
  },
  hooks: {
    afterDelete: ({ existingItem = {} }) => fileAdapter.delete(existingItem),
  },
});
```

I couldn't test the Cloudinary adapter's `delete` function; I just put it together from reading their docs.

### CODE REVIEW

**Comment by MadeByMike** [comment]
This is great I love it :)

File: packages/file-adapters/lib/local-file.js [hunk 1]
```diff
@@ -50,6 +50,24 @@ module.exports = class LocalFileAdapter {
...
+  /**
+   * Takes a file data structure such as existingItem in a hook callback.
+   */
+  delete({ file }) {
+    if (file) {
+      try {
+        fs.unlinkSync(path.join(this.src, file.filename));
+      } catch (err) {
+        // -4058 is the 'no such file or directory' error.
+        if (err.errno === -4058) {
```
Comment (timleslie) [inline packages/file-adapters/lib/local-file.js]: We should probably use `err.code === "ENOENT"` here rather than checking the numerical value.

**Comment by Vultraz** [comment]
BTW, wanted to ask - do you think its better to use `unlink` or `unlinkSync` here? I went with the latter just in case there's a situation with a new file being uploaded with the same name.

**Comment by Vultraz** [comment]
@MadeByMike uhhh how/where do I write a test for this? 😬

File: packages/file-adapters/lib/local-file.js (hunk 1 above)
Comment (MadeByMike) [inline packages/file-adapters/lib/local-file.js]:
```suggestion
        if (err.code === "ENOENT") {
```

**Pull request merged**